src/vm/runners/builtin_runner/keccak.rs

```diff
@@ -0,0 +1,606 @@
+use crate::bigint;
+use crate::hint_processor::builtin_hint_processor::cairo_keccak::keccak_hints::{
+    maybe_reloc_vec_to_u64_array, u64_array_to_mayberelocatable_vec,
+};
+use crate::math_utils::safe_div_usize;
+use crate::types::instance_definitions::keccak_instance_def::KeccakInstanceDef;
+use crate::types::relocatable::{MaybeRelocatable, Relocatable};
+use crate::vm::errors::memory_errors::MemoryError;
+use crate::vm::errors::runner_errors::RunnerError;
+use crate::vm::vm_core::VirtualMachine;
+use crate::vm::vm_memory::memory::Memory;
+use crate::vm::vm_memory::memory_segments::MemorySegmentManager;
+use num_bigint::BigInt;
+use num_integer::div_ceil;
+
+#[derive(Debug, Clone)]
+pub struct KeccakBuiltinRunner {
+    ratio: u32,
+    pub base: isize,
+    pub(crate) cells_per_instance: u32,
+    pub(crate) n_input_cells: u32,
+    verified_addresses: Vec<Relocatable>,
+    pub(crate) stop_ptr: Option<usize>,
+    _included: bool,
+    state_rep: Vec<u32>,
+    instances_per_component: u32,
+}
+
+impl KeccakBuiltinRunner {
+    pub(crate) fn new(instance_def: &KeccakInstanceDef, included: bool) -> Self {
+        KeccakBuiltinRunner {
+            base: 0,
+            ratio: instance_def._ratio,
+            n_input_cells: 25,
```

Comment:
Shouldn't this be `n_input_cells: safe_div(instance_def.cells_per_builtin(), 2)`